src/tiled/abstractobjecttool.cpp

```diff
@@ -246,6 +247,74 @@ void AbstractObjectTool::removeObjects()
     mapDocument()->removeObjects(mapDocument()->selectedObjects());
 }
 
+/**
+ * @brief AbstractObjectTool::applyCollisionsToSelection
+ *
+ * Add the selected collision masks for the currently selected tile - the last selected tile - to
+ * all selected tiles.
+ */
+void AbstractObjectTool::applyCollisionsToSelection()
+{
+    // The selected Collision Masks
+    QList<MapObject*> selectedObjects = mapDocument()->selectedObjects();
```

Comment:
Since we're not going to modify the list of selected objects, we can avoid the extra reference count (and detaching, as will happen when iterating the non-const implicit copy later on):
```c++
const QList<MapObject*> &selectedObjects = mapDocument()->selectedObjects();
```